### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by noelfoka**
## Set up a "Hello world" project as one app with Webpack

In this branch, I:

- [ ] Create a new Rails app called 'hello-rails-react'.
- [ ] Verify that the jsbundling-rails gem is added to your Gemfile.
- [ ] Initialize your project with Git.
- [ ] Install needed node dependencies as described in the previous lesson's tutorial.
- [ ] Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- [ ] Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- [ ] Create a table for storing your messages and create 5 different greetings.
- [ ] Create an API endpoint that selects a random greeting from my table .
- [ ] Create a static view that will be the root of my app.
- [ ] Create the App component with react-router and render it in my static view.
- [ ] Create the Greeting component that will display a greeting. Set it up as a route in my App component.
- [ ] Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- [ ] Display the random greeting in my Greeting component.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Rails.application.routes.draw do
+  get 'api/random_greeting'
+  root 'home#index'
+
+  namespace :api do
+    get '/random_greeting', to: 'api#random_greeting'
+  end
```
Comment (FrederickMih) [inline config/routes.rb]: - [Optional] Let's keep the codebase clean by removing unnecessary codes :+1:

File: app/javascript/redux/greetings/greetingSlice.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
```
Comment (FrederickMih) [inline app/javascript/redux/greetings/greetingSlice.js:1]: - [ ] App doesn't compile successfully due to missing modules in the package.json file. Let's kindly double-check app and make sure it is working properly :+1:
Screenshot from 2023-12-01 00-45-22

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @noelfoka, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Correct config files and required gems have been used :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @noelfoka 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.
✔️ App Run Successfully
image
 

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

**Pull request merged**